### DISCUSSION TITLE
Has anyone fixed code blocks for RTL language pages?

### DISCUSSION THREAD

**Discussion by dogweather** [imfing/hextra#347] [General]
Here's the problem: when the page is, say, Hebrew, the codeblocks are also (incorrectly) shown in RTL:

<img width="689" alt="Screenshot 2024-04-01 at 16 15 45" src="https://github.com/imfing/hextra/assets/150670/4c798e0b-7d67-4450-b972-965530dfc615">

I was able to fix this in the Ananke theme with this:

`layouts/_default/_markup/render-codeblock.html`

```golang
{{ $class := .Attributes.class | default "" }}
{{ $lang  := .Attributes.lang  | default .Type }}

{{ if transform.CanHighlight $lang }}
    <div dir="ltr" class="{{ $class }}">{{ highlight .Inner $lang }}</div>
{{else}}
    <pre dir="ltr"><code class="{{ $class }}">{{.Inner}}</code></pre>
{{end}}
```

This resolves the RTL problem, but it breaks the Hextra styling. It worked with Ananke, so I'm not why this doesn't work with Hextra:

<img width="698" alt="Screenshot 2024-04-01 at 16 23 24" src="https://github.com/imfing/hextra/assets/150670/6d839591-3f22-4119-a824-687d3975e8ce">

**Comment by dogweather** [comment]
I solved it, by making a custom `layouts/partials/components/codeblock.html`. I just added the `dir="ltr"` attributes:

```golang
{{ $filename := .filename | default "" -}}
{{ $lang := .lang | default "" }}
{{ $content := .content }}
{{ $options := .options | default (dict) }}

{{- if $filename -}}
  <div class="filename" dir="auto">{{ $filename }}</div>
{{- end -}}
{{- if transform.CanHighlight $lang -}}
  <div dir="ltr">{{- highlight $content $lang $options -}}</div>
{{- else -}}
  <pre dir="ltr"><code>{{ $content }}</code></pre>
{{- end -}}
```

It's an interesting use case: This is needed for all programming languages shown in codeblocks. Could this be a useful PR for the theme?

<img width="694" alt="Screenshot 2024-04-01 at 16 50 39" src="https://github.com/imfing/hextra/assets/150670/2f0deeb1-0137-4d08-b0e6-97625f60ddbd">

**Comment by imfing** [comment]
not sure but might be related to this pr https://github.com/imfing/hextra/pull/301

**Reply by dogweather** [reply]
Thanks! @Goudarz what would you think about this change, hard-coding all codeblocks `dir="ltr"`? Or should it be an option? Does anyone ever display code RTL?
